src/libopensc/gp.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * gp.c: Global Platform Related functions
+ *
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Author: Jakub Jelen <jjelen@redhat.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "internal.h"
+
+/* The AID of the Card Manager defined by Open Platform 2.0.1 specification */
+static const unsigned char gp_card_manager[7] = {0xA0, 0x00, 0x00, 0x00, 0x03, 0x00, 0x00};
+
+/* The AID of the Issuer Security Domain defined by GlobalPlatform 2.3.1 specification. */
+static const unsigned char gp_isd_rid[7] = {0xA0, 0x00, 0x00, 0x01, 0x51, 0x00, 0x00};
+
+
+/* Select AID */
+static int
+gp_select_aid(struct sc_card *card, const unsigned char *aid)
+{
+	struct sc_apdu apdu;
+
+	sc_format_apdu(card, &apdu, SC_APDU_CASE_3_SHORT, 0xA4, 0x04, 0x0C);
+	apdu.lc = sizeof(aid);
```

Comment:
this won't work. you'll have to pass a length or use `struct sc_aid`

Reply:
True. This got lost in the transition. Thank you for the review. This is fixed now.